public/app/features/org/NewOrgPage.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { FC } from 'react';
+import { getBackendSrv } from '@grafana/runtime';
+import Page from 'app/core/components/Page/Page';
+import { Forms } from '@grafana/ui';
+import { getConfig } from 'app/core/config';
+import { StoreState } from 'app/types';
+import { hot } from 'react-hot-loader';
+import { connect } from 'react-redux';
+import { NavModel } from '@grafana/data';
+import { getNavModel } from '../../core/selectors/navModel';
+
+const createOrg = (newOrg: { name: string }) => {
+  getBackendSrv()
+    .post('/api/orgs/', newOrg)
+    .then((result: any) => {
+      getBackendSrv()
+        .post('/api/user/using/' + result.orgId)
+        .then(() => {
+          window.location.href = getConfig().appSubUrl + '/org';
+        });
+    });
+};
```

Comment:
nit: this could be moved to a thunks action to make it easier to write tests for and separate it a bit more from the component.

Reply:
I'd love to get a walk through on that!

Reply:
![yes!](https://i.giphy.com/media/3OpNkNdOgPBr8kCy6h/giphy.webp)

Reply:
Wonder if we could start thinking about drawing the line (or completely erasing it and go all in Redux) between regular async/await vs thunks...

In this case I don't see need for thunks as it will arguably simplify the code and make it easier to test (think about what's there to test - is it the page component that has strong dependency on this particular async fn, is it the async fn or maybe the form itself that we want to test?). 

 But I see a case for it if we say we make a decision we do all things in Redux...